### ISSUE TITLE
Should Active Model's `ImmutableString` type observe the `limit` parameter?

### ISSUE THREAD

**Issue by freesteph** [rails/rails#51494]
TL;DR: should `attribute :foobar, :string, limit: 30` chop the value to 30 characters?

`ActiveModel::Type::String` inherits
`ActiveModel::Type::ImmutableString` inherits
 `ActiveModel::Type::Value`  which accepts three `limit, scale, precision` options [but lets the base type decide what to do with them](https://api.rubyonrails.org/classes/ActiveModel/Type/Value.html#method-c-new).

I was wondering whether it would make sense for the string types to interpret `limit` as a max length and chop on `cast` or `serialize`. I've got some custom initializer code at the moment which does:

```rb
class ActiveModel::Type::String
  def cast(value)
    if @limit
      super(value.to_s.first(@limit))
    else
      super
    end
  end
end
 ```

Which reuses the `ImmutableString#cast` logic and chops the value, not great but works well, well enough to ask the question here.

What do people think? Happy to submit a PR if we can agree it's nice to have.

NOTE: `ActiveModel::Type::Integer` interprets `limit` as [the number of bytes to store the integer with](https://api.rubyonrails.org/classes/ActiveModel/Type/Integer.html), which makes the option a bit confusing. However, a `t.string :column_name, limit: 42` will effectively [limit the length of that string column in Active Record](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_column).
